### PR TITLE
[Java][Spring] option whether to generate required args constructor

### PR DESCRIPTION
**Pull request by borsch**
Fix for https://github.com/OpenAPITools/openapi-generator/issues/14934

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

### CODE REVIEW

File: docs/generators/java-camel.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ These options may be applied as additional-properties (cli) or configOptions (pl
 |ensureUniqueParams|Whether to ensure parameter names are unique in an operation (rename parameters that are not).| |true|
 |enumUnknownDefaultCase|If the server adds new enum cases, that are unknown by an old spec/client, the client will fail to parse the network response.With this option enabled, each enum will have a new case, 'unknown_default_open_api', so that when the server sends an enum case that is not known by the client/spec, they can safely fallback to this case.|<dl><dt>**false**</dt><dd>No changes to the enum's are made, this is the default option.</dd><dt>**true**</dt><dd>With this option enabled, each enum will have a new case, 'unknown_default_open_api', so that when the enum case sent by the server is not known by the client/spec, can safely be decoded to this case.</dd></dl>|false|
 |fullJavaUtil|whether to use fully qualified name for classes under java.util. This option only works for Java API client| |false|
+|generatedConstructorWithRequiredArgs|Whether to generate constructors with required args for models| |false|
```
Comment (welshm) [inline docs/generators/java-camel.md:59]: Hmm... I wonder if the default should be `true`.

Your use case with Lombok is interesting, but I would think by default developers should populate all required params when constructing a DTO/POJO - so I think this should default to true and if you want to use custom Lombok annotations, you can set to false?

What do you think?

**Review by welshm — APPROVED** [review]
One point of discussion about the default for the option, but code looks good

File: docs/generators/java-camel.md (hunk 1 above)
Comment (borsch) [reply docs/generators/java-camel.md:59]: @welshm from that perspective you are correct, but I'm thinking more from backward compatible perspective. I'm totally fine to change this to opposite value

Comment (borsch) [reply docs/generators/java-camel.md]: @welshm I've changed default value to `true` and I've marked this change as "Breaking with fallback" so we should be fine from release notes perspective

**Pull request merged**